goth/runner/log_monitor.py

```diff
@@ -115,30 +114,53 @@ def _create_file_logger(config: LogConfig) -> logging.Logger:
     return logger_
 
 
-class LogEventMonitor(EventMonitor[LogEvent]):
+class PatternMatchingWaitableMonitor(WaitableEventMonitor[E]):
```

Comment:
If we decide to merge `WaitableEventMonitor` into its base class we can drop the `Waitable` part from this name.

Reply:
Drop, or replace with `Event`